pkg/deploy/controller/deployer/factory.go

```diff
@@ -217,6 +217,7 @@ func (c *DeploymentController) getByKey(key string) (*v1.ReplicationController,
 	}
 	rc, err := c.rcLister.ReplicationControllers(namespace).Get(name)
 	if errors.IsNotFound(err) {
+		// TODO tnozicka: this is not normal and should be refactored
```

Comment:
I've created https://github.com/openshift/origin/issues/15928 for you to address this.